lib/tree.rb

```diff
@@ -1,39 +1,63 @@
 class NoApplesError < StandardError; end
 
-class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+class Tree
+    attr_reader :height, :age, :age_max, :apples, :alive
+
+    def initialize
+        @age = 0
+        @age_max = Random.rand(50..100)
+        @alive = true
+        @apples = []
+        @height = 0
+    end
 
-  def initialize
-  end
+    def age!
+        @age += 1
 
-  def age!
-  end
+        unless @age >= @age_max
```

Comment:
It's slightly easier to read `if` ... `else` (with the two parts swapped) over `unless` ... `else`.

[RuboCop](https://github.com/bbatsov/rubocop) can catch a bunch of this kind of stylistic stuff.